### DISCUSSION TITLE
Spacy 3.0 - specify my own candidate generator to use custom UMLS path

### DISCUSSION THREAD

**Discussion by phil-oxenberg** [explosion/spaCy#7458] [Language Support]
I'm in the process of updating my python project from using Spacy 2 to use Spacy 3. My project uses my own versions of Scispacy candidate_generation.py and linking_utils.py in order to use a custom version of concept_aliases.json and other UMLS data files. How do I accomplish this in Spacy 3?

In V2, I simply made copies of candidate_generation.py and linking_utils.py which referenced local copies of the KB data files and instantiated my version of candidate_generation.py and passed the object to the EntitlyLinker constructor like so:
```
_candidate_generator = CandidateGenerator()
_el = EntityLinker(resolve_abbreviations=True, name="umls",
                                        candidate_generator=_candidate_generator)
```
My candidate_generation.py, in turn, references my linking_utils.py.

**Comment by svlandeg** [comment]
Hi! This seems very scispacy-specific, so perhaps it would make more sense to ask advice [there](https://github.com/allenai/scispacy/)?

**Reply by phil-oxenberg** [reply]
Moved to [here]( https://github.com/allenai/scispacy/issues/337)
